pandas/core/groupby/groupby.py

```diff
@@ -1020,6 +1020,11 @@ def curried(x):
                 return self.apply(curried)
 
             is_transform = name in base.transformation_kernels
+
+            # Transform needs to keep the same schema, including when empty
+            if is_transform and self._obj_with_exclusions.empty:
+                return self._obj_with_exclusions
```

Comment:
I _think_ this will return a DataFrame for a SeriesGroupBy when we want a Series.

Reply:
I did some testing and looks like we do indeed get a Series back; this does get hit by SeriesGroupBy though so still should be tested.